tests/integration/conftest.py

```diff
@@ -240,6 +240,23 @@ def fail_on_errors():
         yield app
 
 
+def fail_missing_secrets(msg):
+    """Fail or skip the test because of missing secrets.
+
+    Secrets are only guaranteed to be available on CI when the branch being tested is
+    committed to the main getsentry repository.  Especially for forks they are hidden to
+    stop external contributors from stealing the secrets.
+
+    This will call `pytest.skip` of the secrets are missing but this is acceptable because
+    e.g. we're on a developer's machine or on a forked repository.  If running on CI in the
+    main repository this will call `pytest.skip`.
+    """
+    if os.getenv("CI") and os.getenv("GITHUB_BASE_REF"):
```

Comment:
according to the docs:

> The base_ref or target branch of the pull request in a workflow run. This property is only available when the event that triggers a workflow run is a pull_request.

> The head_ref or source branch of the pull request in a workflow run. This property is only available when the event that triggers a workflow run is a pull_request.

hm, maybe you can check if base_ref and head_ref are from the same repository? Otherwise I think `base_ref` will always be set also for PRs from forks…

Can you maybe just try this by printing out these vars and see what they are?

Reply:
The docs say (https://docs.github.com/en/free-pro-team@latest/actions/reference/environment-variables):
> GITHUB_HEAD_REF | Only set for forked repositories. The branch of the head repository.
> GITHUB_BASE_REF | Only set for forked repositories. The branch of the base repository.

So I think just presence is enough.  But ugh, the logic is wrong, if it's present it should skip, if not fail.

If you are not forked I think you only get `GITHUB_REF`.